include/libpmemobj++/transaction.hpp

```diff
@@ -446,6 +446,38 @@ class transaction {
 		transaction::run(pool, tx, locks...);
 	}
 
+	/**
+	 * Takes a “snapshot” of a persistent memory block of given size,
+	 * located at the given address ptr in the virtual memory space and
+	 * saves it to the undo log. The application is then free to directly
+	 * modify the object in that memory range. In case of a failure or
+	 * abort, all the changes within this range will be rolled back. The
+	 * supplied block of memory has to be within the pool registered in the
+	 * transaction. This function must be called during TX_STAGE_WORK.
+	 *
+	 * @param[in] addr address of the virtual memory space to be
+	 * snapshotted.
+	 * @param[in] num size of memory block to be snapshotted.
+	 *
+	 * @pre current transaction stage must be equal to TX_STAGE_WORK.
+	 * @post when excpetion is thrown, transaction stage is set to
+	 * TX_STAGE_ONABORT.
+	 *
+	 * @throw transaction_error when snapshotting failed or current
+	 * transaction stage is not equal to TX_STAGE_WORK.
+	 */
+	static void
+	snapshot(void *addr, size_t num)
```

Comment:
definitely const, probably void->char, because this is not safe in C++ and we want people to explicitly cast

we also need to say something about it in the documentation above...